### DISCUSSION TITLE
Cancel running backup?

### DISCUSSION THREAD

**Discussion by jlazic** [ClickHouse/ClickHouse#47566] [General]
I can't find command that would cancel already running async backup.
Let's say I run backup on large table and realize that was bad idea, I see no way to
cancel active backup run.

**Comment by jlazic** [comment]
Forgot to mention that killing backup query/process with `KILL QUERY` does not work, as kill query just waits until backup is finished.

**Comment by marcelbirkner** [comment]
Hi Josip,
only restarting **clickhouse-server** can cancel an already running backup. Currently we don't have a special command for that, though we have it in plans to allow canceling running backups.

**Reply by chayuto** [reply]
Looking forward to this feature as well

**Reply by tvorogme** [reply]
+1

**Comment by AVMusorin** [comment]
This works since [v24.1.1](https://github.com/ClickHouse/ClickHouse/releases/tag/v24.1.1.2048-stable)

```
KILL QUERY WHERE query_kind = 'Backup' AND query_id = '<backup_query_id>'
```

See https://github.com/ClickHouse/ClickHouse/pull/58804
